Sming/appinit/user_main.cpp

```diff
@@ -119,11 +119,11 @@ extern "C" void ICACHE_FLASH_ATTR  __attribute__((weak)) user_pre_init(void)
 
 	enum flash_size_map sizeMap = system_get_flash_size_map();
 	if(!system_partition_table_regist(partitions, ARRAY_SIZE(partitions), sizeMap)) {
-		os_printf(_F("system_partition_table_regist: failed\n"));
-		os_printf(_F("size_map = %u\n"), sizeMap);
+		os_printf("system_partition_table_regist: failed\n");
```

Comment:
Just out of interested: Is the optimized os_printf(X) doing the same or similar to os_printf(_F(X))?

Reply:
Yes. When I looked at other instances of os_printf() (in LWIP especially) I realised it would be simpler redefining `os_printf()` instead, but someone else had already thought of it! Format strings are emitted to `.irom.text` (ICACHE_RODATA_ATTR) rather than `.irom0.text` but end up in the same place.